### DISCUSSION TITLE
MAPPO

### DISCUSSION THREAD

**Discussion by berttggg** [Toni-SM/skrl#109] [Q&A] [answered]
I am trying to deploy skrl mappo into Orbit.
Now I am modifying the Orbit environment, I am wondering about the shape of the data I should send to skrl and the shape of the data I will receive from skrl. 

For example, for `observation_spaces`, will its shape be (num_agents, num_envs, num_observations)?

So, I hope I can know the shape of the following variables to ease my development.

Data sent to skrl:
1. `observation_spaces`
2. `action_spaces`
3. `shared_observation_spaces`
4. reward (in what kind of shape should I send my reward to skrl?)
5. observation (in what kind of shape should I send my observation to skrl?)

Data from skrl:
1. actions (I should expect the actions in what shape?)

Thank you in advance!

**Comment by Toni-SM** [comment, marked as answer]
Hi @berttggg 

The following discussion address the multi-agent environment creation for skrl: https://github.com/Toni-SM/skrl/discussions/102#discussioncomment-6690398

Note that the expected API for multi-agent environment follows the [PettingZoo Parallel API](https://pettingzoo.farama.org/api/parallel/).
